app/models/vacancy.rb

```diff
@@ -133,11 +144,15 @@ def can_receive_job_applications?
   end
 
   def allow_key_stages?
-    !one_phase? || readable_phases == %w[primary] || readable_phases == %w[middle]
+    phases.any? { |phase| phase.in? %w[primary middle secondary through] }
+  end
+
+  def key_stages_for_phases
+    phases.map { |phase| PHASES_TO_KEY_STAGES_MAPPINGS[phase.to_sym] }.flatten.uniq.sort
   end
 
   def allow_subjects?
-    readable_phases != ["primary"]
+    phases != ["primary"]
```

Comment:
Do we need to include `"nursery"` here? It used to be mapped to `"primary"`

Reply:
Thank you, it should be addressed now